src/biome/data/sources/datasource.py

```diff
@@ -77,7 +77,7 @@ def __init__(
 
     @classmethod
     def add_supported_format(
-        cls, format_key: str, parser: Callable, default_params: Dict[str, Any]
+        cls, format_key: str, parser: Callable, default_params: Dict[str, Any] = {}
```

Comment:
Since dicts are mutable, it is a bit dangerous to define them as default value in a method: https://effbot.org/zone/default-values.htm
What you normally do in these cases is:
```suggestion
        cls, format_key: str, parser: Callable, default_params: Dict[str, Any] = None
```
and then inside the method:
```
default_params = default_params or {}
```